pkg/networking/client.go

```diff
@@ -0,0 +1,54 @@
+package networking
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+)
+
+// Client is a diagnostics client for DC/OS.
+type Client struct {
+	http *httpclient.Client
+}
+
+// NewClient creates a new networking client.
+func NewClient(baseClient *httpclient.Client) *Client {
+	return &Client{
+		http: baseClient,
+	}
+}
+
+// IPs returns the private and related public IPs of the nodes in a cluster.
+func (c *Client) IPs() (map[string][]string, error) {
```

Comment:
I'd put this function outside of the client, as it's custom logic used in `dcos node list` to match the table entries with the endpoint information.